src/main.cpp

```diff
@@ -5319,42 +5514,211 @@ bool static ProcessMessage(CNode* pfrom, std::string strCommand, CDataStream& vR
+    }
+    // BUIP010 Xtreme Thinblocks: begin section
+    else if (strCommand == NetMsgType::GET_XTHIN && !fImporting && !fReindex) // Ignore blocks received while importing
+    {
+        networkMessage = new GetXThinMessage(pfrom, chainparams, xthinEnabled);
+    }
+    else if (strCommand == NetMsgType::XTHINBLOCK  && !fImporting && !fReindex) // Ignore blocks received while importing
+    {
+        networkMessage = new XThinBlockMessage(pfrom, xthinEnabled);
+    }
+    else if (strCommand == NetMsgType::XBLOCKTX && !fImporting && !fReindex) // handle Re-requested thinblock transactions
+    {
+        networkMessage = new XBlockTxMessage(pfrom, xthinEnabled);
+    }
+    else if (strCommand == NetMsgType::GET_XBLOCKTX && !fImporting && !fReindex) // return Re-requested xthinblock transactions
+    {
+        networkMessage = new GetXBlockTxMessage(pfrom, xthinEnabled);
+    }
+    // BUIP010 Xtreme Thinblocks: end section
+    else if (strCommand == NetMsgType::BLOCK && !fImporting && !fReindex) // Ignore blocks received while importing
+    {
+        networkMessage = new BlockMessage(pfrom);
+    }
+    else if (strCommand == NetMsgType::GETADDR)
+    {
+        networkMessage = new GetAddrMessage(pfrom);
+    }
+    else if (strCommand == NetMsgType::MEMPOOL)
+    {
+        networkMessage = new MemPoolMessage(pfrom);
+    }
+    else if (strCommand == NetMsgType::PING)
+    {
+        networkMessage = new PingMessage(pfrom);
+    }
+    else if (strCommand == NetMsgType::PONG)
+    {
+        networkMessage = new PongMessage(pfrom);
+    }
+    else if (strCommand == NetMsgType::FILTERLOAD)
+    {
+        networkMessage = new FilterLoadMessage(pfrom);
+    }
+    else if (strCommand == NetMsgType::FILTERADD)
+    {
+        networkMessage = new FilterAddMessage(pfrom);
+    }
+    else if (strCommand == NetMsgType::FILTERCLEAR)
+    {
+        networkMessage = new FilterClearMessage(pfrom);
+    }
+    else if (strCommand == NetMsgType::REJECT)
+    {
+        networkMessage = new RejectMessage(pfrom);
+    }
+    else if (strCommand == NetMsgType::XPEDITEDREQUEST)
+    {
+        networkMessage = new ExpeditedRequestMessage(pfrom);
+    }
+    else if (strCommand == NetMsgType::XPEDITEDBLK)
+    {
+        networkMessage = new ExpeditedBlockMessage(pfrom);
+    }
+    else if (strCommand == NetMsgType::VERSION2)
+    {
+        networkMessage = new Version2Message(pfrom);
+    }
+    else if (strCommand == NetMsgType::VERACK2)
+    {
+        networkMessage = new VersionAck2Message(pfrom);
+    }
     else
     {
         // Ignore unknown commands for extensibility
         logDebug(Log::Net) << "Unknown command" << SanitizeString(strCommand) << "from peer:" << pfrom->id;
     }
 
-
-
-    return true;
+    // FIX-ME: Perhaps a try/catch here according to Ticket 224 description by @zander
+    return networkMessage->handle(vRecv, nTimeReceived, strCommand);;
```

Comment:
Minor nit; double semicolon here.

Major nit; networkMessage may be null here.

Reply:
will fix all the nit stuff and continue